src/main/kotlin/sh/hsp/techtree/YamlReader.kt

```diff
@@ -1,8 +1,20 @@
 package sh.hsp.techtree
 
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import java.io.File
 import java.net.URL
 
 // Wiliniak
 interface YamlReader {
-    fun readToModel(resource: URL): TreeModel
+    fun readToModel(resource: URL): TreeModel{
+        val mapper = ObjectMapper(YAMLFactory()).registerKotlinModule()
```

Comment:
Podałbym objectMapper w konstruktorze zamiast tworzyć i rejestrować przy każdym wywołaniu `readToModel()` (chociaż tutaj to i tak będzie jednorazowe)